lib/auth/helpers.go

```diff
@@ -289,10 +290,22 @@ func NewTestAuthServer(cfg TestAuthServerConfig) (*TestAuthServer, error) {
 		return nil, trace.Wrap(err)
 	}
 
-	authPreference, err := types.NewAuthPreferenceFromConfigFile(types.AuthPreferenceSpecV2{
+	authPreferenceSpec := types.AuthPreferenceSpecV2{
 		Type:         constants.Local,
 		SecondFactor: constants.SecondFactorOff,
-	})
+	}
+
+	if modules.GetModules().Features().Cloud {
```

Comment:
I still think this is surprising, we have what amounts to static state drastically changing the behavior of the test cluster. I think it's better to revert this file and have each test do the prefs update, if they so require it.

Reply:
I have added this because of "cannot disable two-factor authentication on Cloud" error. This happened when I first enable the Cloud feature and then call the web suite:
```
modules.SetTestModules(t, &modules.TestModules{
    TestFeatures: modules.Features{
	Cloud: true,
    },
})
s := newWebSuite(t) // error "cannot disable two-factor authentication on Cloud"
```
This is happening because `newWebSuite > NewTestAuthServer` tries to create an invalid auth preference for the cloud:
```
types.AuthPreferenceSpecV2{
	Type:         constants.Local,
	SecondFactor: constants.SecondFactorOff,
}
```

Reply:
The other tests take another approach:
```go
s := newWebSuite(t)

modules.SetTestModules(t, &modules.TestModules{
    TestFeatures: modules.Features{
	Cloud: true,
    },
})
```
This allows them to bypass the error but if there is some other Cloud code inside `newWebSuite`
```go
if modules.GetModules().Features().Cloud { 
    ...
}
``` 
it will not execute.

Reply:
In my case, I need to make `h.ClusterFeatures.GetCloud()` `true` so I need to pass 
```go
handler, err := NewHandler(Config{
		ClusterFeatures:  *modules.GetModules().Features().ToProto(),
```
ClusterFeatures into handler (in a similar way it is done in production).
This code executes inside `newWebSuite(t)` call so I need to set 
```go
modules.SetTestModules(t, &modules.TestModules{
    TestFeatures: modules.Features{
	Cloud: true,
    },
})
```
before that

Reply:
btw I spotted something confusing. We have two ways to check for Cloud:
 1. `modules.GetModules().Features().Cloud` 
 2. `h.ClusterFeatures.GetCloud()`
What is the preferred way and when?

Reply:
Assuming changing the config after initialization is not feasible, then I think the preferred approach would be to pass in the AuthPreferenceV2Spec we want via the config objects.

As for `modules` vs `h.ClusterFeatures`, I'm not sure. It looks to me that `h.ClusterFeatures` would be preferred, but I wish we had docs to confirm that. Maybe someone who does more in lib/web knows?

Reply:
done in https://github.com/gravitational/teleport/pull/17834/commits

Reply:
Nicely done, thanks.